src/api/namespaces/atomicmarket/handlers/sales2.get-sales-templates.test.ts

```diff
@@ -0,0 +1,423 @@
+
+    describe('getSalesTemplatesAction', () => {
+
+        txit('filters by settlement symbol', async () => {
+            const {template_id} = await initTest();
+            await client.createToken({token_symbol: 'TOKEN1'});
+            const {offer_id} = await client.createOfferAsset({}, {template_id});
+            const {sale_id} = await client.createSale({offer_id, settlement_symbol: 'TOKEN1'});
+
+            expect(await getSalesIds({symbol: 'TOKEN1', burned: false}))
+                .to.deep.equal([sale_id]);
+        });
+
+        txit('filters by minimum price', async () => {
+            const {template_id} = await initTest({
+                listing_price: 100000000,
+            });
+
+            const {offer_id} = await client.createOfferAsset({}, {template_id});
+            const {sale_id} = await client.createSale({
+                offer_id,
+                listing_price: 200000000,
+            });
+
+            expect(await getSalesIds({min_price: '2', burned: false}))
+                .to.deep.equal([sale_id]);
+        });
+
+        txit('filters by maximum price', async () => {
+            const {template_id} = await initTest({
+                listing_price: 300000000,
+            });
+
+            const {offer_id} = await client.createOfferAsset({}, {template_id});
+            const {sale_id} = await client.createSale({
+                offer_id,
+                listing_price: 200000000,
+            });
+
+            expect(await getSalesIds({max_price: '2', burned: false}))
+                .to.deep.equal([sale_id]);
+        });
+
+        txit('filters by collection', async () => {
+            const {template_id} = await initTest();
+
+            const {collection_name} = await client.createCollection({collection_name: 'x'});
+
+            const {offer_id} = await client.createOfferAsset({}, {template_id, collection_name});
+            const {sale_id} = await client.createSale({offer_id, collection_name});
+
+            expect(await getSalesIds({collection_name: 'x,abc'}))
+                .to.deep.equal([sale_id]);
+        });
+
+        txit('filters by asset_id', async () => {
+            const {template_id} = await initTest();
+
+            const {asset_id, offer_id} = await client.createOfferAsset({}, {template_id});
+            const {sale_id} = await client.createSale({offer_id});
+
+            expect(await getSalesIds({asset_id}))
+                .to.deep.equal([sale_id]);
+        });
+
+        txit('only returns single asset sales', async () => {
+            const {template_id, sale_id} = await initTest();
+
+            const {offer_id} = await client.createOfferAsset({}, {template_id});
+            await client.createOfferAsset({offer_id, index: 2}, {
+                template_id: (await client.createTemplate()).template_id,
+            });
+
+            await client.createSale({offer_id});
+
+            expect(await getSalesIds({burned: false}))
```

Comment:
I dont understand this test, wont be better to include a filter for this case and the normal case will be to find all the sales independent of the number of sales?

Reply:
This is for bundled sales (more than 1 asset in a sale).
It's very unlikely that a bundle is cheaper than a single asset, so sales with more than 1 asset are filtered out to make the search faster.